micro_sam/util.py

```
@@ -990,7 +993,12 @@ def segmentation_to_one_hot(segmentation: np.ndarray, segmentation_ids: Optional
         n_ids = int(segmentation.max())
 
     else:
-        assert segmentation_ids[0] != 0, "No objects were found."
+        msg = "No foreground objects were found."
+        if len(segmentation_ids) == 0:  # The list should not be completely empty.
+            raise AssertionError(msg)
```

I would raise a different error type.